pkg/platform/abstract/platform.go

```diff
@@ -308,11 +308,24 @@ func (ap *Platform) EnrichLabels(ctx context.Context, labels map[string]string)
 		labels[common.NuclioResourceLabelKeyProjectName] = platform.DefaultProjectName
 		ap.Logger.DebugCtx(ctx, "No project name specified. Setting to default")
 	}
+	ap.enrichUsernameAndDomainLabels(ctx, labels)
+}
 
+func (ap *Platform) enrichUsernameAndDomainLabels(ctx context.Context, labels map[string]string) {
 	// enrich labels with iguazio.com/username of the creating user
 	if authSession, ok := ctx.Value(auth.AuthSessionContextKey).(*auth.IguazioSession); ok {
-		if value, exist := labels[iguazio.IguzioUsernameLabel]; !exist || value == "" {
-			labels[iguazio.IguzioUsernameLabel] = authSession.GetUsername()
+		if value, exist := labels[iguazio.IguazioUsernameLabel]; !exist || value == "" {
+			fullUsername := authSession.GetUsername()
+
+			// If the username contains '@' (email) then parse the username into two labels: "iguazio.com/username", "iguazio.com/domain"
+			// If it doesn't contain '@' then put the full username into the "iguazio.com/username" label
```

Comment:
instead of writing the code in words, comment why
```suggestion
			// split email usernames to name and domain because '@' is an invalid character in kubernetes labels
```